sockeye/inference.py

```diff
@@ -1051,6 +1067,9 @@ def _beam_search(self,
                 scores = scores[:1] / self.length_penalty(lengths[:1])
             else:
                 # renormalize scores by length ...
+
+                # TODO hussed: why keep multiplying back the LP added in the
```

Comment:
its for renormalization since we are always carrying all hypotheses (including finished ones) through. This is being discussed in the pruning PR.

Lets remove the TODOs that ask questions and discuss them in this PR.